test/e2e/specs/editor/various/block-locking.spec.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'Block Locking', () => {
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+
+	test.describe( 'General', () => {
+		test( 'can prevent removal', async ( { editor, page } ) => {
+			await editor.insertBlock( { name: 'core/paragraph' } );
+			await page.keyboard.type( 'Some paragraph' );
+
+			await editor.clickBlockOptionsMenuItem( 'Lock' );
+
+			await page.click( 'role=checkbox[name="Prevent removal"]' );
```

Comment:
Nit: Ideally, we normally want to add the `i` flag to indicate case insensitive names:

```diff
-await page.click( 'role=checkbox[name="Prevent removal"]' );
+await page.click( 'role=checkbox[name="Prevent removal"i]' );
```

Reply:
I was wondering about the flag since I saw it in other tests. Most labels can be considered case insensitive, should we include this into migration or best practices docs?

I also ended up using `.toBe()` instead of `.toMatchInlineSnapshot()`. Is this also recommended way to test small pieces of content?

Reply:
> Most labels can be considered case insensitive, should we include this into migration or best practices docs?

That would be nice! However, we're still using the unofficial [role-selector](https://github.com/kevin940726/role-selector) created by me. After we switch to the official [role selector](https://playwright.dev/docs/selectors#role-selector) when it becomes stable, we can just omit the `"` to make it more ergonomic.

```diff
-await page.click( 'role=checkbox[name="Prevent removal"i]' );
+await page.click( 'role=checkbox[name=Prevent removal]' );
```

We can do this in one go though.

> I also ended up using `.toBe()` instead of `.toMatchInlineSnapshot()`. Is this also recommended way to test small pieces of content?

Yes, I don't think it's currently possible to use `toMatchInlineSnapshot()` in playwright test runner. So we either have to use `.toMatchSnapshot()` or just `.toBe()`.

Reply:
> After we switch to the official [role selector](https://playwright.dev/docs/selectors#role-selector) when it becomes stable, we can just omit the " to make it more ergonomic.

I thought it became stable in [v1.22.0](https://github.com/microsoft/playwright/releases/tag/v1.22.0).

Reply:
Oh awesome! Didn't catch that. I think we can migrate it to the official selector then :)